contracts/wrappers/algebra-v1.2/PositionManagerAbstractAlgebraV1_2.sol

```diff
@@ -0,0 +1,442 @@
+      !protocolConfig.isTokenEnabled(_token1)
+    ) revert ErrorLibrary.TokenNotEnabled();
+
+    (address token0, address token1) = _getTokensInPoolOrder(_token0, _token1);
+
+    // Deploy and initialize the position wrapper.
+    ERC1967Proxy positionWrapperProxy = new ERC1967Proxy(
+      protocolConfig.getPositionWrapperBaseImplementation(protocolId),
+      abi.encodeWithSelector(
+        IPositionWrapper.init.selector,
+        address(this),
+        token0,
+        token1,
+        _name,
+        _symbol
+      )
+    );
+
+    IPositionWrapper positionWrapper = IPositionWrapper(
+      address(positionWrapperProxy)
+    );
+
+    // Set init values for the position wrapper
+    positionWrapper.setIntitialParameters(0, _tickLower, _tickUpper);
+
+    // Register the new wrapper in the deployed position wrappers list and mark it as a valid wrapper.
+    deployedPositionWrappers.push(address(positionWrapper));
+    externalPositionStorage.addWrappedPosition(address(positionWrapper));
+
+    emit NewPositionCreated(address(positionWrapper), _token0, _token1);
+
+    return positionWrapper;
+  }
+
+  /**
+   * @dev Initializes the position and deposits tokens into it while taking care of dust returns.
+   * @param _dustReceiver Address to send any excess tokens.
+   * @param _positionWrapper Wrapper contract of the position.
+   * @param params Parameters for the position including amounts and ticks.
+   */
+  function _initializePositionAndDeposit(
+    address _dustReceiver,
+    IPositionWrapper _positionWrapper,
+    WrapperFunctionParameters.PositionMintParamsAlgebra memory params
+  ) internal {
+    address token0 = _positionWrapper.token0();
+    address token1 = _positionWrapper.token1();
+
+    // Record balances of token0 and token1 before the transfer to calculate dust later.
+    uint256 balance0Before = IERC20Upgradeable(token0).balanceOf(address(this));
+    uint256 balance1Before = IERC20Upgradeable(token1).balanceOf(address(this));
+
+    // Transfer the specified amounts of token0 and token1 from the sender to this contract.
+    _transferTokensFromSender(
+      token0,
+      token1,
+      params._amount0Desired,
+      params._amount1Desired
+    );
+
+    uint256 balance0After = IERC20Upgradeable(token0).balanceOf(address(this));
+    uint256 balance1After = IERC20Upgradeable(token1).balanceOf(address(this));
+
+    params._amount0Desired = balance0After - balance0Before;
+    params._amount1Desired = balance1After - balance1Before;
+
+    // Mint the new Uniswap V3 position using the provided liquidity parameters.
+    (uint256 tokenId, uint128 liquidity) = _mintNewUniswapPosition(
+      _positionWrapper,
+      params
+    );
+
+    // Set the token ID of the newly minted Uniswap V3 position in the wrapper.
+    _positionWrapper.setTokenId(tokenId);
+
+    // Mint wrapper tokens equivalent to the amount of liquidity added to the Uniswap position.
+    _positionWrapper.mint(msg.sender, liquidity);
```

Comment:
Should this be minted to deployer?

Reply:
See comment above.. the deployer is only relevant for the creation of the external position